### ISSUE TITLE
[RL] vLLM attention masks life cycle

### ISSUE THREAD

**Issue by tianyu-l** [pytorch/torchtitan#3572]
Here're my observations

- `VLLMModelWrapper` doesn't receive `attention_masks` or pass it to transformer layers https://github.com/pytorch/torchtitan/blob/main/torchtitan/experiments/rl/models/vllm_wrapper.py
- the `inner_attention` is swapped from torchtitan `VarlenAttention` / `FlexAttention` to `VLLMAttentionWrapper`, which doesn't receive / pass `attention_masks` to `vllm.model_executor.layers.attention.Attention` https://github.com/pytorch/torchtitan/blob/main/torchtitan/experiments/rl/models/attention.py#L203
- [`BatchInvariantFlexConverter`](https://github.com/pytorch/torchtitan/blob/main/torchtitan/experiments/rl/config_registry.py#L40) only converts torchtitan FlexAttention args, so won't take effect on vllm-side [`attention_config`](https://github.com/pytorch/torchtitan/blob/main/torchtitan/experiments/rl/actors/generator.py#L325)
- It is impossible to make gpt-oss [custom Flex attention masks](https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/gpt_oss/model.py#L217) work, which is beyond document block-causal masking.

In general I wonder how vLLM handles attention masks life cycle.

**Comment by liangel-02** [comment]
> [BatchInvariantFlexConverter](https://github.com/pytorch/torchtitan/blob/main/torchtitan/experiments/rl/config_registry.py#L40) only converts torchtitan FlexAttention args, so won't take effect on vllm-side [attention_config](https://github.com/pytorch/torchtitan/blob/main/torchtitan/experiments/rl/actors/generator.py#L325)

at least right now, BatchInvariantFlexConverter is just matching the vllm attention config (ie BLOCK_M/BLOCK_N are pinned to 16 in vLLM batch invariant mode so we match on the trainer side

**Comment by tianyu-l** [comment]
@liangel-02 
For Flex, do we have plan to support masking besides position-based document masks?

**Comment by liangel-02** [comment]
@tianyu-l yeah, we can pass in custom mask mod to vLLM `FlexAttentionMetadata`, enabled in this PR https://github.com/vllm-project/vllm/pull/37692

**Comment by tianyu-l** [comment]
@liangel-02 
Thanks, do you have plan to enable this on torchtitan side?
